File: leetcode/implementation/942. DI String Match/942. DI String Match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int sz = s.size();
        int l = 0;
        int r = sz;
        vector <int> ans;
        for(int i = 0; i < sz; i++) {
            if(s[i] == 'I') {
                ans.push_back(l++);
            }
            if(s[i] == 'D') {
                ans.push_back(r--);
            }
        }
        ans.push_back(l);
        return ans;
    }
};

File: leetcode/implementation/942. DI String Match/942. DI String Match.rs
impl Solution {
    pub fn di_string_match(s: String) -> Vec<i32> {
        let sz = s.len();
        let mut l = 0;
        let mut r = sz as i32;
        let mut ans:Vec<i32> = Vec::new();
        println!("{:?}",s.chars());
        for c in s.chars() {
            if c == 'I' {
                ans.push(l);
                l += 1;
            } else {
                ans.push(r);
                r -= 1;
            }
        }
        ans.push(l);
        return ans;
    }
}